spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,295 @@ def set_tabs_test(option)
+            1   session1  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "does not return matches when only one field matches" do
+        core.cmd_sessions("--search", "last_checkin:after:1m40.5s session_type:something")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          No matching sessions.
+        TABLE
+      end
+    end
+
+    context "searches for checkin with sessions that do not respond to checkin" do
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, type: 'meterpreter', sid:1, sname: "session1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+        }
+      end
+
+      it "returns no matches when searching against sessions without a checkin" do
+        core.cmd_sessions("--search", "last_checkin:after:6s")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          No matching sessions.
+        TABLE
+      end
+    end
+
+    context "with other flags" do
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "session1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          2 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:2, sname: "session2", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          3 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:3, sname: "session3", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel")
+        }
+      end
+      it "Killall should not kill anything if nothing matches" do
+        core.cmd_sessions("--search", "session_id:5", "-K")
+        expect(@combined_output).to eq([
+          'No matching sessions.'
+        ])
+      end
+
+      it "Killall should kill only the specified sessionS" do
+        expect(sessions[2]).to receive(:kill)
+        core.cmd_sessions("--search", "session_id:2 session_id:3", "-K")
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Killing matching sessions...
+          Active sessions
+          ===============
+
+            Id  Name      Type         Information  Connection
+            --  ----      ----         -----------  ----------
+            2   session2  meterpreter  info         tunnel (127.0.0.1)
+            3   session3  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+    end
+  end
+
+  describe "#parse_duration" do
+    expected_values = {
+          "1s" => 1,
+          "2s" => 2,
+          "1.5m" => 90,
+          "1.5d" => 129600,
+          "1d1h1m1s" => 90061,
+          "1.5d1.5h1.5m1.5s" => 135091,
+          "1.75m70s" => 175
+    }
+    
+    it_behaves_like "parses time values correctly", expected_values
```

Comment:
This isn't quite how you use `it_behaves_like`, which is more useful for testing the different behavior of instances/objects

> Shared examples let you describe behaviour of classes or modules. When declared, a shared group’s content is stored. It is only realized in the context of another example group, which provides any context the shared group needs to run.
> https://rspec.info/features/3-12/rspec-core/example-groups/shared-examples/ 

Paired with the above knowledge - and that it's generally best practice is to have 1 expectation for each `it` block - https://www.betterspecs.org/#single

This pattern might be cleaner:

```suggestion
    {
      "1s" => 1,
      "2s" => 2,
      "1.5m" => 90,
      "1.5d" => 129600,
      "1d1h1m1s" => 90061,
      "1.5d1.5h1.5m1.5s" => 135091,
      "1.75m70s" => 175
    }.each do |input, expected|
      it "returns #{expected} seconds for the input #{input}" do
         expect(core.parse_duration(input)).to eq(output)
      end
    end
```

As it allows you to run failed tests individually from the command line, whilst with your current pattern that's not possible